packages/react/src/data-connect/useConnectMutation.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  useMutation,
+  useQueryClient,
+  type UseMutationOptions,
+  type UseMutationResult,
+} from "@tanstack/react-query";
+import { type FirebaseError } from "firebase/app";
+import {
+  executeMutation,
+  type MutationRef,
+  type DataConnect,
+  type QueryRef,
+} from "firebase/data-connect";
+
+type UseConnectMutationOptions<
+  TData = unknown,
+  TError = FirebaseError,
+  Variables = unknown
+> = Omit<UseMutationOptions<TData, TError, Variables>, "mutationFn"> & {
+  invalidate?: QueryRef<unknown, unknown>[];
```

Comment:
I'll bring this one up, but I'm thinking we should try to support both a QueryRef & the inferred function function where we can just take the name.